code/game/objects/effects/effect_system/effects_smoke.dm

```diff
@@ -263,8 +263,9 @@
 /datum/effect_system/smoke_spread/chem/New()
 	..()
 	chemholder = new /obj()
-	var/datum/reagents/R = new/datum/reagents(500)
+	var/datum/reagents/R = new/datum/reagents(500, INSTANT_REACT)//This is so it can't create infinites
```

Comment:
Can you elaborate on this comment? This alone isn't obvious to me.

Reply:
`//This is a safety for now to prevent smoke generating more smoke as the smoke reagents react in the smoke. This is prevented naturally from happening even if this is off, but I want to be sure that any edge cases are prevented before I get a chance to rework smoke reactions (specifically adding water or reacting away stablising agent in the middle of it).`

It doesn't have to be set like this, but I don't want to worry about this causing problems on live until I come back to revise how this works. It just has the potential for infinite loops which is why I'm overly cautious about it.